src/rdkafka_assignor.h

```diff
@@ -41,6 +41,7 @@ typedef struct rd_kafka_group_member_s {
         rd_kafkap_str_t                 *rkgm_group_instance_id;
         rd_kafkap_bytes_t               *rkgm_userdata;
         rd_kafkap_bytes_t               *rkgm_member_metadata;
+        int                              rkgm_generation;
```

Comment:
this field doesn't really belong here, because it's specific to the sticky assignor. i expect this is best deserialized in `rd_kafka_sticky_assignor_assign_cb` (or helper) and passed as a parameter where needed.

also, this afaict, it's not being set (will always be -1).

also, you don't seem to be parsing the userdata at all - probably better if the owned partitions come from the user data in the EAGER case, even though it won't have any practical downside in the librdkafka case because the sticky partitioner is being introduced alongside the kip429 updates to the rebalance protocol.

Reply:
we could probably skip providing a 'sticky' assignor and only do 'cooperative-sticky'. i doubt anyone will want the former (and it'll be easy to add if requested). this is probably what you thinking i'm now guessing. except `rkgm_generation` should still be just be a constant, as noted above.

Reply:
I wish to keep this to make our sticky assignor implementation as identical as possible to AbstractStickyAssignor.java, I think it is a bigger risk to factor out the generation parts of our implementation than to simply set this to its default value of -1.
It also makes a future demand for the non-cooperative sticky assignor easier to implement.